sopel/tools/target.py

```diff
@@ -23,23 +23,23 @@ class User:
     :param str host: the user's hostname ("host.name" in `user@host.name`)
     """
     __slots__ = (
-        'nick', 'user', 'host', 'realname', 'channels', 'account', 'away',
+        'nick', 'user', 'host', 'realname', 'channels', 'account', 'away', 'is_bot',
     )
 
     def __init__(
         self,
         nick: identifiers.Identifier,
-        user: Optional[str],
-        host: Optional[str],
+        user: str,
+        host: str,
```

Comment:
I'm pretty confident that we have use-cases where we create an empty shell of a user, without `user` and/or `host`. I recommend keeping these as `Optional[str]` for now.

Reply:
You know, you're right. There are a couple places where those values are passed in as `None`. Will revert.